src/arduino/boardManager.ts

```diff
@@ -228,6 +229,11 @@ export class BoardManager {
                     // });
                     if (addedPlatform.name === plat.name) {
                         addedPlatform.versions.push(plat.version);
+                        // Check if this is the latest version. Packages typically support more boards in later versions.

```

Comment:
It's pretty clear what this does in terms of code. In terms of actual implementation, what package is this comparing against to try and make sure all the supported boards are present?

Reply:
I'm not quite sure what you're asking. [Packages are an Arduino concept](https://arduino.github.io/arduino-cli/0.19/package_index_json-specification/) that we consume.

Reply:
I updated the comment to be more precise that we are looking for the latest version of the platform instead of the package as a whole.

Reply:
I guess I'm just confirming that, what this does is make sure that we have all the boards that are supported by confirming that we grab the most recently supported package?

Reply:
Right, a platform will typically (but not necessarily) support more boards in later versions. Unless the user selects otherwise, installing a platform will install the latest version. Selecting boards from the latest version improves search in the board manager because we can typically surface more results.

This isn't guaranteed to represent all the boards that a platform supports, because a platform could use a versioning scheme that we don't understand, or a platform could remove support for certain boards in later versions. However, that's an acceptable tradeoff, and it [matches the way the Arduino IDE works](https://github.com/arduino/Arduino/blob/57a931c9c4c084057417837239ad8136f8a7b1aa/app/src/cc/arduino/contributions/packages/ui/ContributedPlatformReleases.java#L68).